code/modules/clothing/clothing.dm

```diff
@@ -12,6 +12,12 @@
 	var/list/valid_accessory_slots
 	var/list/restricted_accessory_slots
 
+/obj/item/clothing/Destroy()
+	for(var/obj/item/clothing/accessory/A in accessories)
+		qdel(A)
+	accessories = null
```

Comment:
Should this not be accessories.Cut() if the intent is to free up references?

Reply:
We null the reference to the list itself. And the "Cut" part happens a row above.

Reply:
Though I think there is a define like QDEL_LIST()